tests/unit/models/test_evc_deploy.py

```diff
@@ -1102,131 +1110,89 @@ def create_evc_intra_switch():
-        )
-        evc.uni_z.user_tag = uni_z_tag
-
-        # Test3: no TAG in both UNI_Z and UNI_Z
-        evc.uni_a.user_tag = None
-        evc.uni_z.user_tag = None
-        expected_flows_no_tag = [
-            {
-                "match": {"in_port": 1},
-                "cookie": evc.get_cookie(),
-                "actions": [{"action_type": "output", "port": 3}],
-                "priority": EPL_SB_PRIORITY
-            },
-            {
-                "match": {"in_port": 3},
-                "cookie": evc.get_cookie(),
-                "actions": [{"action_type": "output", "port": 1}],
-                "priority": EPL_SB_PRIORITY
-            }
-        ]
-        evc._install_direct_uni_flows()
-        send_flow_mods_mock.assert_called_with(
-            expected_dpid, expected_flows_no_tag
-        )
-#
-#        print(evc._prepare_direct_uni_flows())
-#        evc.uni_a.user_tag = uni_a_tag
-#        uni_z_tag = evc.uni_z.user_tag
-#        evc.uni_z.user_tag = None
-#        print(evc._prepare_direct_uni_flows())
-#        evc.uni_z.user_tag = uni_z_tag
-#        evc.uni_a.user_tag = None
-#        evc.uni_z.user_tag = None
-#        print(evc._prepare_direct_uni_flows())
-#        self.assertTrue(False)
+        for uni_a in [100, "4096/4096", 0, None]:
+            for uni_z in [100, "4096/4096", 0, None]:
+                expected_flows = [
+                    {
+                        "match": {"in_port": 1},
+                        "cookie": evc.get_cookie(),
+                        "actions": [
+                            {"action_type": "output", "port": 3},
+                        ],
+                        "priority": EPL_SB_PRIORITY
+                    },
+                    {
+                        "match": {"in_port": 3},
+                        "cookie": evc.get_cookie(),
+                        "actions": [
+                            {"action_type": "output", "port": 1},
+                        ],
+                        "priority": EPL_SB_PRIORITY
+                    }
+                ]
+                evc.uni_a = get_uni_mocked(tag_value=uni_a, is_valid=True)
+                evc.uni_a.interface.port_number = 1
+                evc.uni_z = get_uni_mocked(tag_value=uni_z, is_valid=True)
+                evc.uni_z.interface.port_number = 3
+                expected_dpid = evc.uni_a.interface.switch.id
+                evc._install_direct_uni_flows()
+                if uni_a is not None:
+                    expected_flows[0]["match"]["dl_vlan"] = uni_a
+                if uni_z is not None:
+                    expected_flows[1]["match"]["dl_vlan"] = uni_z
+
+                if uni_z not in {None, "4096/4096", 0}:
+                    expected_flows[1]["priority"] = EVPL_SB_PRIORITY
+                    expected_flows[0]["actions"].insert(
+                        0, {"action_type": "set_vlan", "vlan_id": uni_z}
+                    )
+                elif uni_z == "4096/4096":
```

Comment:
- You can use a func to set the priority accordingly here too that was mentioned in the review. Could you use it here too to simplify?
- Could you also instead of `for` statements with `assertions` transform this into `pytest` `parametrize` or `with self.subTest`? parametrizing `uni_a` and `uni_z` accordingly? `parametrize` decorator is preferred, but since `mef_eline` still has tech debt using `unittest.TestCase` (and this is incompatible with `parametrize`) then it's OK to use `subTest` too whichever is easier.

Reply:
Looks good. Let me close this.